### ISSUE TITLE
TypeError: len() of unsized object

### ISSUE THREAD

**Issue by chrisfilo** [poldracklab/fmriprep#981]
https://openneuro.org/datasets/ds001224/versions/00001?app=FMRIPREP&version=49&job=5a80a195831753000151c639

```
180212-17:57:59,100 workflow WARNING:
	 [Node] Error on "fmriprep_wf.single_subject_262_wf.func_preproc_task_walkstatement_run_03_wf.bold_confounds_wf.ica_aroma_wf.ica_aroma" (/root/src/fmriprep/work/fmriprep_wf/single_subject_262_wf/func_preproc_task_walkstatement_run_03_wf/bold_confounds_wf/ica_aroma_wf/ica_aroma)
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 11, in <module>
    load_entry_point('fmriprep==1.0.6', 'console_scripts', 'fmriprep')()
  File "/usr/local/miniconda/lib/python3.6/site-packages/fmriprep/cli/run.py", line 267, in main
    fmriprep_wf.run(**plugin_settings)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/workflows.py", line 602, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/base.py", line 168, in run
    self._clean_queue(jobid, graph, result=result))
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/base.py", line 227, in _clean_queue
    raise RuntimeError("".join(result['traceback']))
RuntimeError: Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 68, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 487, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 571, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 650, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 516, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/interfaces/report_base.py", line 54, in _run_interface
    ReportCapableInterface, self)._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 1023, in _run_interface
    self.raise_exception(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 960, in raise_exception
    ).format(**runtime.dictcopy()))
RuntimeError: Command:
ICA_AROMA.py -den nonaggr -i /root/src/fmriprep/work/fmriprep_wf/single_subject_262_wf/func_preproc_task_walkstatement_run_03_wf/bold_confounds_wf/ica_aroma_wf/smooth/vol0000_xform-00000_merged_smooth.nii.gz -meldir /root/src/fmriprep/work/fmriprep_wf/single_subject_262_wf/func_preproc_task_walkstatement_run_03_wf/bold_confounds_wf/ica_aroma_wf/melodic -mc /root/src/fmriprep/work/fmriprep_wf/single_subject_262_wf/func_preproc_task_walkstatement_run_03_wf/bold_hmc_wf/normalize_motion/motion_params.txt -o /root/src/fmriprep/work/fmriprep_wf/single_subject_262_wf/func_preproc_task_walkstatement_run_03_wf/bold_confounds_wf/ica_aroma_wf/ica_aroma
Standard output:

------------------------------- RUNNING ICA-AROMA ------------------------------- 
--------------- 'ICA-based Automatic Removal Of Motion Artifacts' --------------- 

Step 1) MELODIC
  - The existing/specified MELODIC directory will be used.
  - The MELODIC directory does not contain the required 'stats' folder. Mixture modeling on the Z-statistical maps will be run.
Step 2) Automatic classification of the components
  - registering the spatial maps to MNI
  - extracting the CSF & Edge fraction features
     - The spatial map of component 1 is empty. Please check!
     - The spatial map of component 2 is empty. Please check!
     - The spatial map of component 3 is empty. Please check!
     - The spatial map of component 4 is empty. Please check!
     - The spatial map of component 5 is empty. Please check!
     - The spatial map of component 6 is empty. Please check!
     - The spatial map of component 7 is empty. Please check!
     - The spatial map of component 8 is empty. Please check!
     - The spatial map of component 9 is empty. Please check!
     - The spatial map of component 10 is empty. Please check!
     - The spatial map of component 11 is empty. Please check!
     - The spatial map of component 12 is empty. Please check!
     - The spatial map of component 13 is empty. Please check!
     - The spatial map of component 14 is empty. Please check!
  - extracting the Maximum RP correlation feature
  - extracting the High-frequency content feature
  - classification
Standard error:
Traceback (most recent call last):
  File "/opt/ICA-AROMA/ICA_AROMA.py", line 211, in <module>
    motionICs = aromafunc.classification(outDir, maxRPcorr, edgeFract, HFC, csfFract)
  File "/opt/ICA-AROMA/ICA_AROMA_functions.py", line 487, in classification
    if len(motionICs) != 0:
TypeError: len() of unsized object
Return code: 1

Error response from daemon: No such container: f269d8d9-081e-46c3-9bcb-102ea3f631ea-lock
No lock found for f269d8d9-081e-46c3-9bcb-102ea3f631ea
```

I don't see any apparent issues with the data and MRIQC-ICA runs without issues.

**Comment by chrisfilo** [comment]
Might be related to https://github.com/maartenmennes/ICA-AROMA/issues/10

**Comment by chrisfilo** [comment]
also reported in https://neurostars.org/t/fmriprep-errors-during-freesurfer-recon-all-or-during-ica-aroma/1472

**Comment by effigies** [comment]
If the issue you linked fixed it, we should upgrade ICA-AROMA to 0.4.3-beta. (Currently at 0.4.1-beta.)

Can we reliably reproduce the error? If so, it would be good to test this before committing to a new version.

**Comment by effigies** [comment]
(Actually, fixed in 0.4.2-beta, but 0.4.3-beta seems to differ by whitespace.)

**Comment by chrisfilo** [comment]
You can use https://openneuro.org/datasets/ds001224/versions/00001?app=FMRIPREP&version=49&job=5a80a195831753000151c639 to reproduce the error. Thanks in advance for looking into this.

**Comment by effigies** [comment]
I was unable to reproduce the error on a subject that exhibited the error in OpenNeuro (sub-262). In the most recent runs, it seems to be timing out on recon-all, not having ICA-AROMA issues. I can try running all subjects to try to reproduce, but do you have any better leads?

**Comment by chrisfilo** [comment]
No other than this could be stochastic. ICA gives slightly different
results each run afaik.

Best,
Chris

PS Apologies for brevity and potential typos. This message was composed on
a phone.

**Comment by effigies** [comment]
Been running that dataset on 1.0.8 all weekend. 89 BOLD series have had ICA-AROMA run on them without error, so far.

I'm not running FreeSurfer, so it's possible there's a bbregister-AROMA interaction.

**Comment by effigies** [comment]
Ran the full dataset, no issues. Will try running a subject with FreeSurfer.

**Comment by ccharpen** [comment]
Hi! Any update on this issue when running subjects with FreeSurfer? Is it possible that there is a bbregister-AROMA interaction?
So far on my side, after finally getting through my dataset of 30 subjects (using fmriprep 1.0.8 with freesurfer and aroma), I am left with 7 subjects who show an ICA-aroma error on one or more runs. For 4 of these subjects the error is the same as described here. However for the remaining 3 subjects, this is the error I am getting: 

> Node: fmriprep_wf.single_subject_028_wf.func_preproc_task_ObsLearn_run_08_wf.ica_aroma_wf.ica_aroma
> Working directory: /home/ccharpen/ObsLearn/prepdata/work/fmriprep_wf/single_subject_028_wf/func_preproc_task_ObsLearn_run_08_wf/ica_aroma_wf/ica_aroma
> 
> Node inputs:
> 
> TR = <undefined>
> args = <undefined>
> compress_report = auto
> denoise_type = nonaggr
> dim = <undefined>
> environ = {}
> feat_dir = <undefined>
> fnirt_warp_file = <undefined>
> generate_report = True
> ignore_exception = False
> in_file = /home/ccharpen/ObsLearn/prepdata/work/fmriprep_wf/single_subject_028_wf/func_preproc_task_ObsLearn_run_08_wf/ica_aroma_wf/smooth/vol0000_xform-00000_merged_smooth.nii.gz
> mask = <undefined>
> mat_file = <undefined>
> melodic_dir = /home/ccharpen/ObsLearn/prepdata/work/fmriprep_wf/single_subject_028_wf/func_preproc_task_ObsLearn_run_08_wf/ica_aroma_wf/melodic
> motion_parameters = /home/ccharpen/ObsLearn/prepdata/work/fmriprep_wf/single_subject_028_wf/func_preproc_task_ObsLearn_run_08_wf/bold_hmc_wf/normalize_motion/motion_params.txt
> out_dir = <undefined>
> out_report = ica_aroma_reportlet.svg
> report_mask = /home/ccharpen/ObsLearn/prepdata/work/fmriprep_wf/single_subject_028_wf/func_preproc_task_ObsLearn_run_08_wf/bold_mni_trans_wf/mask_mni_tfm/ref_image_corrected_brain_mask_maths_trans.nii.gz
> terminal_output = <undefined>
> 
> Traceback (most recent call last):
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 68, in run_node
>     result['result'] = node.run(updatehash=updatehash)
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 487, in run
>     result = self._run_interface(execute=True)
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 571, in _run_interface
>     return self._run_command(execute)
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 650, in _run_command
>     result = self._interface.run(cwd=outdir)
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 516, in run
>     runtime = self._run_interface(runtime)
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/interfaces/report_base.py", line 69, in _run_interface
>     self._generate_report()
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/interfaces/segmentation.py", line 153, in _generate_report
>     noise_components_file=self._noise_components_file
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/viz/utils.py", line 543, in plot_melodic_components
>     noise_components = [int(c) for c in cf.read().split(",")]
>   File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/viz/utils.py", line 543, in <listcomp>
>     noise_components = [int(c) for c in cf.read().split(",")]
> ValueError: invalid literal for int() with base 10: ''

Are the two errors linked?
Both are reproducible (ran 2 or 3 times each and got the same error).

Thanks!

**Comment by ccharpen** [comment]
No it's not sorry - currently running fmriprep on our HPC using singularity container. I think that I am not allowed to share raw, unprocessed data for IRB issues, but I'll double check and get back to you - maybe I could privately send you one of these problematic subjects?

**Comment by effigies** [comment]
Sure. If you're able to, the easiest way to share is to upload the partial dataset to [OpenNeuro](https://openneuro.org/) and you can share the dataset with me (this username @ gmail). Datasets must be defaced and pass [BIDS validation](http://incf.github.io/bids-validator/).

**Comment by oesteban** [comment]
I think this has been fixed with #1096 and #1099. Can we confirm @chrisfilo @ccharpen ?

**Comment by oesteban** [comment]
As per https://neurostars.org/t/fmriprep-errors-during-freesurfer-recon-all-or-during-ica-aroma/1472/8 and other threads this should have been fixed for good. Please reopen if we see a regression.
